python/lsst/analysis/drp/gatherResourceStatistics.py

```diff
@@ -0,0 +1,119 @@
+# This file is part of analysis_drp.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from collections import defaultdict
+
+__all__ = ()
+
+from lsst.pex.config import ListField
+from lsst.pipe.base import PipelineTaskConnections, PipelineTask, PipelineTaskConfig
+from lsst.pipe.base import connectionTypes as ct
+
+
+class GatherResourceStatisticsConnections(PipelineTaskConnections, dimensions=()):
+    output_table = ct.Output(
+        "resource_statistics",
+        storageClass="DataFrame",
+        dimensions=(),
+        doc=(
+            "Table that aggregates memory and CPU usage statistics from one "
+            "or more tasks. "
+            "This will have one row for each data ID, with columns for each "
+            "task or method's memory usage and runtime."
+        )
+    )
+
+    def __init__(self, *, config):
+        super().__init__(config=config)
+        # Inject one new input connection for each configured label.
+        for label in config.grouped_labels():
+            self.inputs.add(label)
+            dataset_type_name = f"{label}_metadata"
```

Comment:
There is a `TaskDef.metadataDatasetName` that forms the `label_metadata` name.  That's used by `ctrl_mpexec`. Should we have a class method on `TaskDef` that can be used rather than hard-coding it in here?